### ISSUE TITLE
Isolate space occupied by numbers and tabs

### ISSUE THREAD

**Issue by trevorhauter** [philmard/tab-numbers#2]
Hey there! I absolutely love this plugin, it was the first thing I looked for when I downloaded zen (which was earlier today), thank you for making this! I noticed the other issue [1] has a complaint about the number overlapping with the tab, more specifically the X button. 

I have a similar issue with the overlapping.

![Screenshot 2024-12-07 at 9 06 54 PM](https://github.com/user-attachments/assets/b1e43434-72d4-41d6-a4fa-ba72a911af40)

 Is it possible to adjust the styling so that the tab takes up a smaller portion of the horizontal width, leaving more room for the number and preventing any overlap? 

Here is, admittedly, a poor ascii illustration of what I am trying to describe. This could potentially resolve both issues, though it look like maybe a solution was already provided for the first.
\____________________
|_GitHub______________X_| $tab_num

1. https://github.com/philmard/tab-numbers/issues/1

**Comment by philmard** [comment]
Hello, and thank you for the kind words!

I think I see what you 're saying. 
You basically want the tab content separated from the Tab Number - **_when the tabs are expanded_** -. 
(When the tabs are compact, the style will mostly remain the same.)

I'm currently working on something like this: 
(example 1: blue background, rectangular)
<img width="227" alt="tabs_blue_rectangular" src="https://github.com/user-attachments/assets/e0c3d48c-12c1-4db8-a519-260cb6687ad1">

and this: 
(example 2: orange background, round)
<img width="230" alt="tabs_orange_circular" src="https://github.com/user-attachments/assets/df859dd3-7caa-4cdb-abec-0d017ff1e48a">

and: 
(example 3: grey, left side)
<img width="230" alt="tabs_grey_left" src="https://github.com/user-attachments/assets/731b1593-bf16-43a9-8907-b11ec9567b10">

The user will have the ability to change 

- the colour of the background, 
- its border-radius, 
- the side in which the Tab Numbers will be, 

etc. through the preferences.

but I think the compact style should mostly stay the same, like so (with the ability to make the tab numbers smaller/bigger):
<img width="51" alt="tabs_compact" src="https://github.com/user-attachments/assets/38c2b2bb-517d-41cb-9148-bd2ff108f3e2">

**Comment by trevorhauter** [comment]
@philmard yup, that addresses everything above and beyond what I hoped. Can't wait!!

**Comment by CosmoCreeper** [comment]
This issue should probably be closed. I think it's fixed now right?

**Comment by philmard** [comment]
@CosmoCreeper  Yes, you're right. I'm closing it!
